weblate/static/editor/full.js

```diff
@@ -111,72 +133,222 @@
             }
-            this.isTMLoaded = true;
+            self.isTMLoaded = true;
+            self.initTranslationMemory();
+        });
+    };
+
+    FullEditor.prototype.initMachineTranslation = function () {
+        var self = this;
+        increaseLoading('mt');
+        $.ajax({
+            url: $('#js-mt-services').attr('href'),
+            success: function (servicesList) {
+                decreaseLoading('mt');
+                servicesList.forEach(function (serviceName) {
+                    increaseLoading('mt');
+                    self.fetchMachinery(serviceName);
+                });
+            },
+            error: self.processMachineryError,
+            dataType: 'json'
+        });
+    };
+
+    FullEditor.prototype.initTranslationMemory = function () {
+        increaseLoading('memory');
+        this.fetchMachinery(TM_SERVICE_NAME);
+
+        var self = this;
+        $('#memory-search').submit(function () {
+            var form = $(this);
+
             increaseLoading('memory');
-            var $form = $('#link-post');
+            $('#memory-translations').empty();
             $.ajax({
                 type: 'POST',
-                url: $('#js-translate').attr('href').replace('__service__', 'weblate-translation-memory'),
+                url: form.attr('action'),
+                data: form.serialize(),
+                dataType: 'json',
                 success: function (data) {
-                    processMachineTranslation(data, 'memory');
+                    self.processMachineryResults(data, 'memory');
                 },
                 error: function (jqXHR, textStatus, errorThrown) {
-                    failedMachineTranslation(jqXHR, textStatus, errorThrown, 'memory');
-                },
-                dataType: 'json',
-                data: {
-                    csrfmiddlewaretoken: $form.find('input').val(),
+                    self.processMachineryError(jqXHR, textStatus, errorThrown, 'memory');
                 },
             });
+            return false;
+        });
+    };
 
-            $('#memory-search').submit(function () {
-                var form = $(this);
-
-                increaseLoading('memory');
-                $('#memory-translations').empty();
-                $.ajax({
-                    type: 'POST',
-                    url: form.attr('action'),
-                    data: form.serialize(),
-                    dataType: 'json',
-                    success: function (data) {
-                        processMachineTranslation(data, 'memory');
-                    },
-                    error: function (jqXHR, textStatus, errorThrown) {
-                        failedMachineTranslation(jqXHR, textStatus, errorThrown, 'memory');
-                    },
-                });
-                return false;
+    FullEditor.prototype.fetchMachinery = function (serviceName) {
+        var serviceType = serviceName === TM_SERVICE_NAME ? 'memory' : 'mt';
```

Comment:
The type depends rather on source where it is triggered - the machine translation tab displays translation memory matches as well, so my guess is that this will break display of the translation memory matches on the machine translation tab.

Reply:
Ah I see, the MT tab also provides TM results... what's the motivation behind it if I may ask? I'd like to understand it because as a translator I find that quite confusing.

Reply:
There are two reasons, but non of them is really good :-).

* There used to be just the MT tab in past and when adding TM tab, the results stayed there.
* Some third-party services are also TM (or even hybrid like MyMemory, which can provide both translation memory matches and machine translation).

Reply:
Ok, I see... I'll make sure not to touch existing behavior but please reconsider this in the future! None of the reasons you mentioned seem like roadblocks not to have a real split.

Reply:
Issue noted: https://github.com/WeblateOrg/weblate/issues/4041